tests/integration/cmd_push_test.go

```diff
@@ -318,7 +316,30 @@ var _ = Describe("odo push command tests", func() {
 			// verify that the README.md file was not pushed
 			stdOut3 := oc.ExecListDir(podName, project, dir)
 			Expect(stdOut3).To(Not(ContainSubstring(("README.md"))))
+		})
+	})
+
+	Context("when .gitignore file exists", func() {
+		It("should create and push the contents of a named component and include odo-file-index.json path to .gitignore file to exclude the contents", func() {
+			helper.CmdShouldPass("git", "clone", "https://github.com/openshift/nodejs-ex", context+"/nodejs-ex")
+			helper.CmdShouldPass("odo", "component", "create", "nodejs", cmpName, "--project", project, "--context", context+"/nodejs-ex", "--app", appName)
+
+			// push and include the odo-file-index.json path to .gitignore file
+			helper.CmdShouldPass("odo", "push", "--context", filepath.Join(context, "nodejs-ex"))
+			ignoreFilePath := filepath.Join(context, "nodejs-ex", ".gitignore")
+			helper.FileShouldContainSubstring(ignoreFilePath, filepath.Join(".odo", "odo-file-index.json"))
```

Comment:
Then you need to verify inside the pod that ```odo-file-index.json``` is not pushed.

Reply:
IMO as it is related to gitignore so while git push it should not be pushed. I think its nowhere related to pod and hence IMO it will not be the used case.

Reply:
ok